OpenRA.Game/Graphics/CursorProvider.cs

```diff
@@ -31,7 +31,7 @@ public CursorProvider(ModData modData)
 
 			// Overwrite previous definitions if there are duplicates
 			var pals = new Dictionary<string, IProvidesCursorPaletteInfo>();
-			foreach (var p in modData.DefaultRules.Actors["world"].TraitInfos<IProvidesCursorPaletteInfo>())
+			foreach (var p in modData.DefaultRules.Actors[SystemActors.World.ToString().ToLowerInvariant()].TraitInfos<IProvidesCursorPaletteInfo>())
```

Comment:
This is worse than hardcoding "world", IMO. Does anybody else have any thoughts about this?

Reply:
Something like `modData.DefaultRules.SystemActor(SystemActors.World).TraitInfos<>()` might be a better compromise.

Reply:
I don't agree this is worse. I would have gone with strings constants just for cleaner syntax, but this is still better than hardcoding "world".
Delegating the work to a centralized spot sounds good. Maybe even define the "special" actor names at the mod level - either in the manifest or in the rules, so they can be accessed similarly to what was suggested without requiring the mod to call them "world" and "player".

Reply:
An even cleaner solution would be to change `modData.DefaultRules.Actors` to a new `ActorInfoDictionary` type that provides a custom indexer keyed on `SystemActors` in addition to strings. This would allow these lines to be written as `modData.DefaultRules.Actors[SystemActors.World].TraitInfos<>()` directly.

You can copy/paste `ConcurrentCache` which gives 95% of the code needed for this.